core/arch/arm/include/kernel/thread_private_arch.h

```diff
@@ -25,7 +25,12 @@
 #else
 #define STACK_TMP_SIZE		(2048 + STACK_TMP_OFFS)
 #endif
+
+#ifdef CFG_WITH_PAGER
+#define STACK_THREAD_SIZE	(8192 + CFG_STACK_THREAD_EXTRA)
```

Comment:
I think `CFG_STACK_THREAD_EXTRA` should be included in `STACK_THREAD_SIZE` regardless of pager configuration. In fact the current code looks buggy, see `print_stack_limits()` / `thread_stack_start()` / `thread_stack_size()` / `get_stack_limits()`. They all ignore `CFG_STACK_THREAD_EXTRA`.

Reply:
I agree (hence why I didnt just add it to init_thread_stacks as it is done on the non pager case)

So based on this, 

```
#ifndef CFG_WITH_PAGER
DECLARE_STACK(stack_thread, CFG_NUM_THREADS,
	      STACK_THREAD_SIZE + CFG_STACK_THREAD_EXTRA, static);
#endif

```

I could have done

```
#ifdef CFG_WITH_PAGER
static void init_thread_stacks(void)
{
	size_t n = 0;

	/*
	 * Allocate virtual memory for thread stacks.
	 */
	for (n = 0; n < CFG_NUM_THREADS; n++) {
		tee_mm_entry_t *mm = NULL;
		vaddr_t sp = 0;
		size_t num_pages = 0;
		struct fobj *fobj = NULL;

		/* Find vmem for thread stack and its protection gap */
		mm = tee_mm_alloc(&tee_mm_vcore,
				  SMALL_PAGE_SIZE + STACK_THREAD_SIZE + CFG_STACK_THREAD_EXTRA );
		assert(mm);
```

but It seemed wrong.

But this code area being so sensitive I was hoping someone with more experience to comment ; to me adding it to STACK_THREAD_SIZE is bullet-proof and easier to test and manage.

should we do that then?

Reply:
Let me try something, I'll share with you so that you can test on STM32MP1 with pager.

Reply:
https://github.com/OP-TEE/optee_os/pull/5500

Reply:
will test this in an hour or two but the patch is pretty much what I had in  mind. nice!

Reply:
@jforissier works for me with the pager on STM32MP1. 
I'll close this PR then